src/miner_gateway_port.erl

```diff
@@ -79,6 +82,25 @@ handle_info({'DOWN', Ref, port, _Pid, Reason}, #state{port = Port, monitor = Ref
                    State#state.tcp_port),
     ok = miner_gateway_ecc_worker:reconnect(),
     {noreply, NewState};
+handle_info(gw_healthcheck, State) ->
+    Me = self(),
+    Healthcheck = fun() -> Output = os:cmd(gateway_bin() ++ " -c " ++ gateway_config_dir() ++ " info"), Me ! Output end,
+    Checker = spawn(Healthcheck),
+    receive
+        Response ->
+            case string:str(Response, "gateway") of
```

Comment:
Would be nice if we could pattern match this instead. Maybe something like:
```erl
case iolist_to_binary(Response) of
    <<"gateway", _/binary>> -> ...;
    <<_/binary>> -> ...
end
```

and even better with `case {ExitCode, iolist_to_binary(Response)} of`

Reply:
we don't really care about the contents of the output so much as that we _get_ the output; the condition we're trying to guard against is the helium_gateway getting hung such that calling the above command never returns _anything_

Reply:
oh, sweet - that's even simpler then - just check exit code.